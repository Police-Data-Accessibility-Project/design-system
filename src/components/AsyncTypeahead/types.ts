export interface PdapAsyncTypeaheadProps<T> {
	id: string;
	placeholder?: string;
	items: T[] | undefined;
	formatItemForDisplay?: (item: T) => string;
	error?: string;
	position?: 'top' | 'bottom';
}
