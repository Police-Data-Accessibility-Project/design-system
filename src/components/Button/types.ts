export interface PdapButtonProps {
	intent?: 'primary' | 'secondary' | 'tertiary';
	isLoading?: boolean;
}
