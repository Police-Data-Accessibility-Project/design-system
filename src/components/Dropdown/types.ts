export enum PdapDropdownTriggerType {
	PRESS = 'press',
	HOVER = 'hover',
}

export interface PdapDropdownProps {
	defaultOpen?: boolean;
	disabled?: boolean;
	triggerOn?: PdapDropdownTriggerType;
}
