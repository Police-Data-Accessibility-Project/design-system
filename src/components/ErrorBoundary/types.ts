import { ComponentPublicInstance } from 'vue';

export interface PdapErrorEmitted {
	error: Error;
	vm: ComponentPublicInstance | null;
	info?: string;
}

export interface PdapErrorBoundaryProps {
	component: string;
	onError?: (
		error: Error,
		target?: ComponentPublicInstance | null | undefined,
		info?: string
	) => void;
	params?: Record<string, string>;
}
