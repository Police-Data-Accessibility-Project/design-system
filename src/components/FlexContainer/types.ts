export interface PdapFlexContainerProps {
	alignment?: 'center' | 'start';
	className?: string;
	component?: string;
}
