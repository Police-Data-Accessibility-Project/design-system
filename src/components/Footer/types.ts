import { FOOTER_LINK_ICONS } from './constants';

export interface PdapFooterSocialLinks {
	href?: string;
	path?: string;
	text: string;
	icon?: FooterIconName;
}
export type FooterIconName =
	(typeof FOOTER_LINK_ICONS)[keyof typeof FOOTER_LINK_ICONS];

export interface PdapFooterProps {
	fundraisingData: {
		raised: number;
		goal: number;
	};
}
