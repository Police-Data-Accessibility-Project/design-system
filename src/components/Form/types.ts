import { PdapInputCheckboxProps, PdapInputTextProps } from '../Input/types';

export type PdapLengthRules = 'maxLength' | 'minLength';

export interface PdapFormValidator<T> {
	message?: string;
	value: T;
}

/**
 * Keyed by currently used validators.
 * Add any Vuelidate validators or custom ones here as we need them.
 * See https://vuelidate-next.netlify.app/validators.html#using-builtin-validators for more.
 *
 */
export interface PdapFormValidators {
	maxLength: PdapFormValidator<number>;
	minLength: PdapFormValidator<number>;
	required: PdapFormValidator<boolean>;
}

export interface PdapFormSchemaEntryInputCheckbox
	extends PdapInputCheckboxProps {
	validators?: Partial<PdapFormValidators>;
}
export interface PdapFormSchemaEntryInputText extends PdapInputTextProps {
	validators?: Partial<PdapFormValidators>;
}
export type PdapFormSchemaEntry =
	| PdapFormSchemaEntryInputCheckbox
	| PdapFormSchemaEntryInputText;
export type PdapFormSchema = PdapFormSchemaEntry[];
export type PdapFormData = Record<string, unknown>;

/**
 * PDAP Form props interface.
 */
export interface PdapFormProps {
	error?: string | undefined | null;
	id: string;
	name: string;
	schema: PdapFormSchema;
}
