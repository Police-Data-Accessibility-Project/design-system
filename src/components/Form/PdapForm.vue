<template>
	<form
		:id="id"
		:name="name"
		class="pdap-form"
		v-bind="$attrs"
		@submit.prevent="submit"
	>
		<div
			v-if="typeof errorMessage === 'string'"
			class="pdap-form-error-message"
		>
			{{ errorMessage }}
		</div>

		<PdapInput
			v-for="field in data"
			v-bind="field"
			:key="field.name"
			:error="
				Boolean(v$[field.name]?.$error)
					? v$[field.name]?.$errors?.[0]?.$message
					: ''
			"
			:value="values[field.name]"
			@input="updateForm(field, $event)"
		/>
		<slot />
	</form>
</template>

<script setup lang="ts">
// Globals
import { computed, ref, watchEffect } from 'vue';
import { useVuelidate } from '@vuelidate/core';

// Components
import PdapInput from '../Input/PdapInput.vue';

// Utils
import { createRule } from '../../utils/vuelidate';

// Types
import { PdapFormProps } from './types';
import { PdapInputProps, PdapInputTypes } from '../Input/types';

// Props
const props = withDefaults(defineProps<PdapFormProps>(), {
	error: null,
	resetOn: 'submit',
});

// Emits
const emit = defineEmits(['submit', 'change']);

// State
const data = computed(() =>
	props.schema.map((input) => {
		const newInput = { ...input };
		delete newInput.validators;

		return newInput;
	})
);

const values = ref<Record<string, string>>(
	data.value.reduce((acc, input) => {
		switch (input.type) {
			case PdapInputTypes.CHECKBOX:
				return { ...acc, [input.name]: String(input.defaultChecked) };
			case PdapInputTypes.TEXT:
			case PdapInputTypes.PASSWORD:
			default:
				return { ...acc, [input.name]: input.value };
		}
	}, {})
);
const rules = props.schema.reduce((acc, input) => {
	const toAdd = Object.entries(input.validators ?? {}).reduce(
		(acc, [key, val]) => {
			return {
				...acc,
				...createRule(key, val),
			};
		},
		{}
	);
	return {
		...acc,
		[input.name]: {
			...toAdd,
		},
	};
}, {});
const v$ = useVuelidate(rules, values, { $autoDirty: false, $lazy: true });

// Vars
const errorMessage = ref(props.error);

// Handlers
function updateForm(field: PdapInputProps, event: Event) {
	const target = event.target as HTMLInputElement;

	const update = (() => {
		switch (field.type) {
			case PdapInputTypes.CHECKBOX:
				return target.checked ? 'true' : 'false';
			default:
				return target.value;
		}
	})();

	values.value[field.name] = update;
	emit('change', values.value, event);
}

// Effects
// Effect - Updates form error state based on input error state and/or props
watchEffect(() => {
	if (props.error) errorMessage.value = props.error;
	else if (errorMessage.value && v$.value.$errors.length === 0)
		errorMessage.value = null;
	else if (!errorMessage.value && v$.value.$errors.length > 0)
		errorMessage.value = 'Please update this form to correct the errors';
});

watchEffect(() => {
	if (props.resetOn && props.resetOn !== 'submit') {
		resetForm();
	}
});

/**
 * Reset vuelidate and wipe values state
 */
function resetForm() {
	v$.value.$reset();
	values.value = Object.entries(values).reduce((acc, [key]) => {
		return { ...acc, [key]: '' };
	}, {});
}

async function submit(e: Event) {
	// Check form submission
	const isValidSubmission = await v$.value.$validate();
	if (isValidSubmission) {
		// Emit submit event (spread to new object to create new object, this allows us to reset `values` without messing with the data returned)
		emit('submit', { ...values.value }, e);

		if (props.resetOn === 'submit') {
			resetForm();
		}
	}
}
</script>

<script lang="ts">
/**
 * # `Form`
 * The `Form` component is powerful. All you need to do is pass a few props, and the component will generate inputs and render them in the UI, complete with customizable form validation and both form-level and input-level error states.
 *
 *
 * ## Props
 * @prop {string | undefined | null} error Error state. Only a non-falsy string results in a form-level error being displayed
 * @prop {string} id Passed through to the `form` element as its `id`
 * @prop {string} name Passed through to the `form` element as its `name`
 * @prop {PdapFormSchema} schema Array of `PdapFormInputProps` objects.
 *
 * ## Notes
 *
 * - Form schema entries can look different depending on the type of input. We currently only use text inputs, so the example only displays these.
 * - To properly submit the form, you must render a button with `type="submit"` _inside_ of the `Form` component.
 * - `Form` emits a `submit` event and passes all values to the handler you pass to `on-submit`
 * - Currently available form validations are defined by the `PdapFormValidators` interface:
 *
 * ```
 * PdapFormValidators {
 *   maxLength: {
 *     message?: string;
 *     value: number;
 *   };
 *   minLength: {
 *     message?: string;
 *     value: number;
 *   };
 *   required: {
 *     message?: string;
 *     value: boolean;
 *   };
 * }
 * ```
 *
 *
 * @example
 *
 * <template>
 *  <Form :schema="formSchema" :on-submit="handleSubmit" id="test-form" name="data-request-form">
 *    <Button intent="primary" type="submit">Click me</Button>
 *  </Form>
 * </template>
 *
 *
 * <st>
 * import { Button, Form, PdapInputTypes } from 'pdap-design-system';
 *
 * export default {
 *  components: ['Button', 'Form'],
 *  data() {
 *    return {
 *      formSchema: [
 *        {
 *          id: 'testfirstname',
 *          name: 'firstName',
 *          label: 'First Name',
 *          type: 'text',
 *          placeholder: 'John',
 *          value: '',
 *          validators: {
 *            minLength: {
 *              value: 3
 *            },
 *            required: {
 *              message: 'Please provide this information',
 *              value: true
 *            }
 *          },
 *        },
 *        {
 *          id: 'testlastname',
 *          name: 'lastName',
 *          label: 'Last Name',
 *          type: 'text',
 *          placeholder: 'Doe',
 *          value: '',
 *          validators: {
 *            minLength: {
 *              value: 3
 *            },
 *            maxLength: {
 *              message: 'A thousand characters for your surname? Are you a bot?',
 *              value: 999
 *            },
 *          },
 *        {
 *          id: 'ice-cream',
 *          name: 'iceCream',
 *          label: 'Do you like ice cream?',
 *          type: 'checkbox',
 *          defaultChecked: true,
 *        }
 *      ]
 *    }
 *  },
 *  methods: {
 *    handleSubmit: async function(data) {
 *      await doRequestStuff(data);
 *      this.$router.push('/')
 *    }
 *  }
 *}
 * </scriptt>
 */
export default {
	name: 'PdapForm',
};
</script>

<style>
@tailwind components;

@layer components {
	.pdap-form {
		@apply mb-4 w-full;
	}

	.pdap-form-error-message {
		@apply items-center justify-start basis-full flex-shrink flex bg-red-300 text-red-800 mb-3 p-2 text-sm;
	}
}
</style>
