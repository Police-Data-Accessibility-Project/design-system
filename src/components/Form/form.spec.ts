// Component
import PdapForm from './PdapForm.vue';

// Utils
import { mount } from '@vue/test-utils';
import { describe, expect, test, vi } from 'vitest';
import { PdapInputTypes } from '../Input/types';
import { nextTick } from 'vue';

const schema = [
	{
		id: 'test-1',
		label: 'First test input',
		name: 'testOne',
		placeholder: 'Enter a value here',
		type: PdapInputTypes.TEXT,
		value: '',
		validators: {
			required: {
				message: 'This field is required',
				value: true,
			},
		},
	},
	{
		id: 'test-email',
		label: 'Email test input',
		name: 'email',
		placeholder: 'Email',
		type: PdapInputTypes.TEXT,
		value: '',
		validators: {
			email: {
				message: 'Please enter a valid email address',
				value: true,
			},
		},
	},
	{
		id: 'test-password',
		label: 'Password test input',
		name: 'password',
		placeholder: 'Password',
		type: PdapInputTypes.PASSWORD,
		value: '',
		validators: {
			password: {
				message:
					'Please enter a password of 8 characters, with at least one capital letter, one lowercase letter, one number, and one special character',
				value: true,
			},
		},
	},
	{
		id: 'test-2',
		label: 'Second test input',
		name: 'testTwo',
		placeholder: 'Enter another value here',
		type: PdapInputTypes.TEXT,
		value: '',
		validators: {
			minLength: {
				message: 'This must be at least two characters',
				value: 2,
			},
		},
	},
	{
		id: 'checkbox-default-checked',
		name: 'checkboxDefaultChecked',
		label: 'Keep this box checked?',
		type: PdapInputTypes.CHECKBOX,
		defaultChecked: true,
	},
	{
		id: 'checkbox-default-unchecked',
		name: 'checkboxDefaultUnchecked',
		label: 'Check this box?',
		type: PdapInputTypes.CHECKBOX,
		defaultChecked: false,
	},
];

vi.mock('vue-router');
vi.mock('vue', async () => {
	const actual: Record<string, unknown> = await vi.importActual('vue');
	return {
		...actual,
		/** Shim for `exportHelper` function which throws errors on Input because actual
		 * Vue impl does not have optional chaining on `sfc.__vccOpts` check */
		exportHelper: function (
			sfc: Record<string, unknown>,
			props: [string, string][]
		) {
			const newObject = sfc?.vccOpts ?? sfc;
			for (const [key, val] of props) {
				newObject[key] = val;
			}
			return newObject;
		},
	};
});
vi.mock('@vuelidate/core', async () => {
	const actual: Record<string, unknown> =
		await vi.importActual('@vuelidate/core');

	return {
		...actual,
	};
});

const submit = vi.fn((values: Record<string, string>) => values);
const base = {
	global: {
		mocks: {
			$route: {
				fullPath: '/',
			},
			$router: {
				push: vi.fn(),
			},
		},
	},
	props: {
		schema,
		id: 'test',
		name: 'test',
	},
	attrs: {
		onSubmit: submit,
	},
	slots: {
		default: '<button type="submit">Submit</button>',
	},
};

describe('Form component', () => {
	test('Renders component in static state', () => {
		const wrapper = mount(PdapForm, base);

		expect(wrapper.find('.pdap-form').exists()).toBe(true);
		expect(wrapper.find('.pdap-form-error-message').exists()).toBe(false);
		expect(wrapper.find('#test-1').exists()).toBe(true);
		expect(wrapper.find('#test-1').attributes('validators')).toBe(undefined);
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('Renders component in form error state', () => {
		const wrapper = mount(PdapForm, {
			...base,
			props: { ...base.props, error: 'This form is incorrect' },
		});

		expect(wrapper.find('.pdap-form').exists()).toBe(true);
		expect(wrapper.find('.pdap-form-error-message').exists()).toBe(true);
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('Renders correct inputs per schema, accepts user actions', async () => {
		const wrapper = mount(PdapForm, base);

		// Get elements
		const inputTextOne = await wrapper.find('#test-1');
		const inputTextTwo = await wrapper.find('#test-2');
		const inputEmail = await wrapper.find('#test-email');
		const inputPassword = await wrapper.find('#test-password');
		const inputCheckboxDefaultChecked = await wrapper.find(
			'#checkbox-default-checked'
		);
		const inputCheckboxDefaultUnchecked = await wrapper.find(
			'#checkbox-default-unchecked'
		);
		await nextTick();

		const elements = [
			inputTextOne,
			inputTextTwo,
			inputEmail,
			inputPassword,
			inputCheckboxDefaultChecked,
			inputCheckboxDefaultUnchecked,
		];

		// Assert existence and expected default values
		elements.forEach((element) => {
			expect(element.exists()).toBe(true);
		});
		elements.slice(0, 3).forEach((element) => {
			expect((element.element as HTMLInputElement).value).toBe('');
		});
		expect(
			(inputCheckboxDefaultChecked.element as HTMLInputElement).value
		).toBe('true');
		expect(
			(inputCheckboxDefaultUnchecked.element as HTMLInputElement).value
		).toBe('false');

		// Interact with elements
		await inputTextOne.setValue('foo');
		await inputTextTwo.setValue('bar');
		await inputEmail.setValue('hello@hello.com');
		await inputPassword.setValue('P@ssword1!');
		await inputCheckboxDefaultChecked.trigger('change');
		await inputCheckboxDefaultUnchecked.trigger('change');
		await nextTick();

		// Assert new values
		expect((inputTextOne.element as HTMLInputElement).value).toBe('foo');
		expect((inputTextTwo.element as HTMLInputElement).value).toBe('bar');
		expect((inputEmail.element as HTMLInputElement).value).toBe(
			'hello@hello.com'
		);
		expect((inputPassword.element as HTMLInputElement).value).toBe(
			'P@ssword1!'
		);
		expect(
			(inputCheckboxDefaultChecked.element as HTMLInputElement).value
		).toBe('true');
		expect(
			(inputCheckboxDefaultUnchecked.element as HTMLInputElement).value
		).toBe('false');
	});

	test('Submits with correct values', async () => {
		const wrapper = mount(PdapForm, base);

		const inputTextOne = wrapper.find('#test-1');
		const inputTextTwo = wrapper.find('#test-2');
		const inputEmail = await wrapper.find('#test-email');
		const inputPassword = await wrapper.find('#test-password');

		await nextTick();

		await inputTextOne.setValue('foo');
		await inputTextTwo.setValue('bar');
		await inputEmail.setValue('hello@hello.com');
		await inputPassword.setValue('P@ssword1!');

		await wrapper.find('form').trigger('submit');
		await nextTick();
		await wrapper.vm.$forceUpdate();

		expect(wrapper.emitted('submit')?.[0][0]).toStrictEqual({
			checkboxDefaultChecked: 'true',
			checkboxDefaultUnchecked: 'false',
			email: 'hello@hello.com',
			password: 'P@ssword1!',
			testOne: 'foo',
			testTwo: 'bar',
		});
		expect(submit).toHaveBeenCalled();
	});

	test('Form validation with incorrect values and clears form error state', async () => {
		const wrapper = mount(PdapForm, base);

		const inputTextOne = wrapper.find('#test-1');
		const inputTextTwo = wrapper.find('#test-2');
		const inputEmail = await wrapper.find('#test-email');
		const inputPassword = await wrapper.find('#test-password');
		await nextTick();

		await inputTextOne.setValue('');
		await inputTextTwo.setValue('a');
		await inputEmail.setValue('hello');
		await inputPassword.setValue('Pass');
		await nextTick();

		await wrapper.find('form').trigger('submit');
		await nextTick();
		wrapper.vm.$forceUpdate();

		// Assert error state
		expect(wrapper.find('.pdap-form-error-message').exists()).toBe(true);
		// Assert error message
		expect(wrapper.find('.pdap-form-error-message').text()).toBe(
			'Please update this form to correct the errors'
		);

		// Set values to correct values
		await inputTextOne.setValue('aaaaa');
		await inputTextTwo.setValue('bvvvvv');
		await inputEmail.setValue('hello@hello.com');
		await inputPassword.setValue('P@ssword1!');
		await nextTick();
		wrapper.vm.$forceUpdate();

		// Assert error message no longer present
		expect(wrapper.find('.pdap-form-error-message').exists()).toBe(false);
	});

	test('Form waits to reset until resetOn prop switches to `true` and error is falsy', async () => {
		const wrapper = mount(PdapForm, {
			...base,
			props: { ...base.props, error: 'foo', resetOn: false },
		});

		// Assert error state
		expect(wrapper.find('.pdap-form-error-message').exists()).toBe(true);
		// Assert error message
		expect(wrapper.find('.pdap-form-error-message').text()).toBe('foo');

		// Set values to correct values
		await wrapper.setProps({ error: '', resetOn: true });

		// Assert error message no longer present
		expect(wrapper.find('.pdap-form-error-message').exists()).toBe(false);
	});
});
