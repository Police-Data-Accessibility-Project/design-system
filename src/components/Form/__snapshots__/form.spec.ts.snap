// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form component > Renders component in form error state 1`] = `
<form class="pdap-form">
  <div class="pdap-form-error-message">This form is incorrect</div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-1" name="testOne" placeholder="Enter a value here" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-1">First test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-email" name="email" placeholder="Email" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-email">Email test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-password" name="password" placeholder="Password" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-password">Password test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-2" name="testTwo" placeholder="Enter another value here" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-2">Second test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <!--v-if-->
    <!-- Password -->
    <!-- Checkbox -->
    <input checked class="pdap-input-checkbox" id="checkbox-default-checked" name="checkboxDefaultChecked" type="checkbox" value="true">
    <!--v-if-->
    <label class="pdap-input-label" for="checkbox-default-checked">Keep this box checked?</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <!--v-if-->
    <!-- Password -->
    <!-- Checkbox -->
    <input class="pdap-input-checkbox" id="checkbox-default-unchecked" name="checkboxDefaultUnchecked" type="checkbox" value="false">
    <!--v-if-->
    <label class="pdap-input-label" for="checkbox-default-unchecked">Check this box?</label>
  </div>
  <button type="submit">Submit</button>
</form>
`;

exports[`Form component > Renders component in static state 1`] = `
<form class="pdap-form">
  <!--v-if-->
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-1" name="testOne" placeholder="Enter a value here" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-1">First test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-email" name="email" placeholder="Email" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-email">Email test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-password" name="password" placeholder="Password" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-password">Password test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <input id="test-2" name="testTwo" placeholder="Enter another value here" type="text"><!-- Password -->
    <!--v-if-->
    <!--v-if-->
    <label class="pdap-input-label" for="test-2">Second test input</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <!--v-if-->
    <!-- Password -->
    <!-- Checkbox -->
    <input checked class="pdap-input-checkbox" id="checkbox-default-checked" name="checkboxDefaultChecked" type="checkbox" value="true">
    <!--v-if-->
    <label class="pdap-input-label" for="checkbox-default-checked">Keep this box checked?</label>
  </div>
  <div class="pdap-grid-container pdap-input">
    <!-- Text -->
    <!--v-if-->
    <!-- Password -->
    <!-- Checkbox -->
    <input class="pdap-input-checkbox" id="checkbox-default-unchecked" name="checkboxDefaultUnchecked" type="checkbox" value="false">
    <!--v-if-->
    <label class="pdap-input-label" for="checkbox-default-unchecked">Check this box?</label>
  </div>
  <button type="submit">Submit</button>
</form>
`;
